### PR TITLE
BUILD-4122 Use digicert tooling to sign

### PR DESCRIPTION
**Pull request by tomverin**
Please review our [contribution guidelines](https://github.com/SonarSource/sonarlint-eclipse/blob/master/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/SLE) ticket available, please make your commits and pull request start with the ticket ID (SLE-XXXX)

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -78,17 +78,26 @@ build_task:
...
+    variable_name: SM_CLIENT_CERT_FILE_BASE64
   <<: *SETUP_MAVEN_CACHE
   build_script: |
+    base64 --decode "${SM_CLIENT_CERT_FILE}.b64" > "${SM_CLIENT_CERT_FILE}"
+    smctl healthcheck
     source cirrus-env BUILD-PRIVATE
-    base64 --decode "${CODESIGNING_JKS_PATH}.b64" > "${CODESIGNING_JKS_PATH}"
-    .cirrus/regular_mvn_build_deploy_analyze -Dmaven.test.skip=true -Dsonar.skip=true
+    echo .cirrus/regular_mvn_build_deploy_analyze -Dmaven.test.skip=true -Dsonar.skip=true
+    mvn clean install -Dmaven.test.skip=true -Dsonar.skip=true -Psign
```
Comment (tomverin) [inline .cirrus.yml:100]: remove this line

File: .cirrus.star [hunk 2]
```diff
@@ -1,4 +1,4 @@
 load("github.com/SonarSource/cirrus-modules@v2", "load_features")
 
 def main(ctx):
-  return load_features(ctx)
+  return load_features(ctx, aws=dict(env_type="dev"))
```
Comment (tomverin) [inline .cirrus.star:4]: Remove this change when the docker images containing digicert tooling are promoted to prod

File: .cirrus/Dockerfile [hunk 3]
```diff
@@ -2,7 +2,7 @@ ARG CIRRUS_AWS_ACCOUNT=275878209202
 ARG JDK_VERSION=17
 FROM public.ecr.aws/docker/library/eclipse-temurin:11-jammy AS buildphase
 
-FROM ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j${JDK_VERSION}-m3.9-latest
+FROM ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base-pr-389:j${JDK_VERSION}-m3.9-latest
```
Comment (julien-carsique-sonarsource) [inline .cirrus/Dockerfile:5]: Remove this change when the docker images containing digicert tooling are promoted to prod

File: .cirrus.yml [hunk 4]
```diff
@@ -78,16 +78,24 @@ build_task:
     SIGN_KEY: VAULT[development/kv/data/sign data.key]
     PGP_PASSPHRASE: VAULT[development/kv/data/sign data.passphrase]
     DEPLOY_PULL_REQUEST: true
-    CODESIGNING_JKS_FILE: VAULT[development/team/sonarlint/kv/data/codesigning/2021-2023 data.file]
-    CODESIGNING_JKS_PATH: ${CIRRUS_WORKING_DIR}/SonarSource-2021-2023.jks
-    CODESIGNING_JKS_PASSWORD: VAULT[development/team/sonarlint/kv/data/codesigning/2021-2023 data.password]
+    #CODESIGNING_JKS_FILE: VAULT[development/team/sonarlint/kv/data/codesigning/2021-2023 data.file]
+    #CODESIGNING_JKS_PATH: ${CIRRUS_WORKING_DIR}/SonarSource-2021-2023.jks
+    #CODESIGNING_JKS_PASSWORD: VAULT[development/team/sonarlint/kv/data/codesigning/2021-2023 data.password]
```
Comment (julien-carsique-sonarsource) [inline .cirrus.yml:83]: to clean up?

File: .cirrus.star (hunk 2 above)
Comment (thahnen) [reply .cirrus.star]: done

File: .cirrus/Dockerfile (hunk 3 above)
Comment (thahnen) [reply .cirrus/Dockerfile]: done

File: .cirrus.yml (hunk 4 above)
Comment (thahnen) [reply .cirrus.yml]: done

File: .cirrus.yml [hunk 5]
```diff
@@ -78,16 +78,21 @@ build_task:
...
+    SM_HOST: VAULT[development/team/sonarlint/kv/data/codesigning/2023-2025 data.host]
   codesigning_file:
-    path: ${CODESIGNING_JKS_PATH}.b64
-    variable_name: CODESIGNING_JKS_FILE
+    path: ${SM_CLIENT_CERT_FILE}.b64
+    variable_name: SM_CLIENT_CERT_FILE_BASE64
   <<: *SETUP_MAVEN_CACHE
   build_script: |
+    base64 --decode "${SM_CLIENT_CERT_FILE}.b64" > "${SM_CLIENT_CERT_FILE}"
+    smctl healthcheck
```
Comment (tomverin) [inline .cirrus.yml:94]: this line can be removed as weel

Comment (thahnen) [reply .cirrus.yml]: done

**Review by thahnen — APPROVED** [review]
LGTM, gonna test it with the dogfooding artifact once again after the PR is closed and the first master build is done.

**Pull request merged**